include/sycl/algorithm/inclusive_scan.hpp

```diff
@@ -107,7 +109,258 @@ OutputIterator inclusive_scan(ExecutionPolicy &sep, InputIterator b,
   q.wait_and_throw();
   return o + vectorSize;
 }
+#else
 
+
+struct mapscan_descriptor {
+  size_t size, size_per_work_group, size_per_work_item,
+    nb_work_group, nb_work_item;
+  mapscan_descriptor(size_t size_,
+                     size_t size_per_work_group_,
+                     size_t size_per_work_item_,
+                     size_t nb_work_group_,
+                     size_t nb_work_item_):
+    size(size_),
+    size_per_work_group(size_per_work_group_),
+    size_per_work_item(size_per_work_item_),
+    nb_work_group(nb_work_group_),
+    nb_work_item(nb_work_item_) {}
+};
```

Comment:
Also, can you add some documentation explaining what the descriptors are and how should they be used?